content_script.js

```diff
@@ -108,17 +108,20 @@ class HTMLSerializer {
     this.html.push(`style=${quote}${style}${quote} `);
 
     var attributes = element.attributes;
+    var processedAttributes = new Set();
     if (attributes) {
       for (var i = 0, attribute; attribute = attributes[i]; i++) {
         switch (attribute.name.toLowerCase())  {
           case 'src':
-            this.processSrcAttribute(element);
+            this.processSrcAttribute(element, processedAttributes);
           case 'style':
             break;
           default:
-            var name = attribute.name;
-            var value = attribute.value;
-            this.processSimpleAttribute(name, value);
+            if (!processedAttributes.has(attribute.name.toLowerCase())) {
```

Comment:
Yes.  The only reason I added the processedAttributes set, is because when I am processing a src attribute, I add height and width attributes as well, and I want to avoid adding them twice. Perhaps there is a simpler way to do this, instead of passing this set around to all of my method calls?